### PR TITLE
build: replace `squirrelly` with `eta`

### PR DESCRIPTION
**Pull request by davidmurdoch**
This PR replaces `squirrelly` (https://www.npmjs.com/package/squirrelly) with `eta` (https://www.npmjs.com/package/eta).

`eta` is smaller, more popular, and more maintained. It is also the default html engine used in our up-and-coming webpack-based build system.

Squirrelly has a 52/100 Package Health Score https://snyk.io/advisor/npm-package/squirrelly
Eta has an 86/100 Package Health Score https://snyk.io/advisor/npm-package/eta

<!--
## **Description**

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/23215?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -20,3 +20,5 @@ app/_locales/** linguist-generated
 # yarn berry suggested .gitattributes
 /.yarn/releases/** binary
 /.yarn/plugins/** binary
+
+app/*.html linguist-language=EJS
```
Comment (davidmurdoch) [inline .gitattributes:24]: This tells the GitHub UI to highlight our `app/*.html` files as if they were `EJS` template files (`eta` uses `ejs` syntax).

File: .vscode/extensions.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
   "recommendations": [
     "rvest.vs-code-prettier-eslint",
-    "streetsidesoftware.code-spell-checker"
+    "streetsidesoftware.code-spell-checker",
+    "DigitalBrainstem.javascript-ejs-support"
```
Comment (davidmurdoch) [inline .vscode/extensions.json:5]: Without this plugin VSCode will not highlight the EJS parts of our HTML template properly (they'll just be HTML, which is fine -- but its nice when the lines are highlighted properly).

with plugin:

image

without: 
image

File: .vscode/settings.json [hunk 3]
```diff
@@ -2,6 +2,9 @@
   "cSpell.words": ["blockaid", "lavamoat"],
   "editor.defaultFormatter": "rvest.vs-code-prettier-eslint",
   "editor.tabSize": 2,
+  "files.associations": {
+    "app/*.html": "ejs"
```
Comment (davidmurdoch) [inline .vscode/settings.json:6]: Tells VSCode that `app/*.html` files are to be treated as EJS template files (it gracefully falls back to HTML if the EJS syntax plugin isn't installed)

File: development/build/scripts.js [hunk 4]
```diff
@@ -1325,7 +1325,8 @@ function renderHtmlFile({
   const htmlFilePath = `./app/${htmlName}.html`;
   const htmlTemplate = readFileSync(htmlFilePath, 'utf8');
 
-  const htmlOutput = Sqrl.render(htmlTemplate, { isMMI, isTest });
+  const eta = new Eta();
+  const htmlOutput = eta.renderString(htmlTemplate, { isMMI, isTest });
```
Comment (davidmurdoch) [inline development/build/scripts.js:1329]: The interface is _almost_ as simple as `squirrelly`; i've asked if they'd add static helpers: https://github.com/eta-dev/eta/issues/275

File: package.json [hunk 5]
```diff
@@ -200,7 +200,6 @@
     "web3@^0.20.7": "patch:web3@npm%3A0.20.7#./.yarn/patches/web3-npm-0.20.7-ee7ef00c57.patch",
     "watchify@^4.0.0": "patch:watchify@npm%3A4.0.0#./.yarn/patches/watchify-npm-4.0.0-4fd965dd49.patch",
     "undeclared-identifiers@^1.1.2": "patch:undeclared-identifiers@npm%3A1.1.2#./.yarn/patches/undeclared-identifiers-npm-1.1.2-13d6792e9e.patch",
-    "squirrelly@^9.0.0": "patch:squirrelly@npm%3A9.0.0#./.yarn/patches/squirrelly-npm-9.0.0-3cf710c7bb.patch",
```
Comment (davidmurdoch) [inline package.json:203]: I love removing patches!

File: .gitattributes (hunk 1 above)
Comment (davidmurdoch) [inline .gitattributes:24]: This is supposed to make GitHub UI treat our `app/*.html` files as if they were EJS templates, but it doesn't seem to be working?

docs are here: https://github.com/github-linguist/linguist/blob/master/docs/overrides.md#using-gitattributes

The docs mention:

> the added attributes will not take effect until the .gitattributes file is committed to your repository.

And maybe that actually means it needs to be committed to the default branch to take effect?

Anyone know what's i'm doing wrong, if anything?

**Review by legobeat — APPROVED** [review]

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**